From c16b48c8980ea73db3d90ad67206f8adc015613b Mon Sep 17 00:00:00 2001
From: Zhao Qiang <qiang.zhao@nxp.com>
Date: Tue, 28 Aug 2018 17:27:20 +0800
Subject: [PATCH 575/767] CAAM: fix compile error

This compile error is caused by backport. In the porting to LTS 4.14,
ivsize was defined in ablkcipher_givencrypt, but it should be in
ablkcipher_decrypt.

Signed-off-by: Zhao Qiang <qiang.zhao@nxp.com>
[Xulin: Original patch taken from NXP LSDK-18.09.]
Signed-off-by: Xulin Sun <xulin.sun@windriver.com>
---
 drivers/crypto/caam/caamalg_qi2.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/crypto/caam/caamalg_qi2.c b/drivers/crypto/caam/caamalg_qi2.c
index 79520f1eac7b..9f7c6c6069c6 100644
--- a/drivers/crypto/caam/caamalg_qi2.c
+++ b/drivers/crypto/caam/caamalg_qi2.c
@@ -1930,7 +1930,6 @@ static int ablkcipher_givencrypt(struct skcipher_givcrypt_request *greq)
 	struct crypto_ablkcipher *ablkcipher = crypto_ablkcipher_reqtfm(req);
 	struct caam_ctx *ctx = crypto_ablkcipher_ctx(ablkcipher);
 	struct caam_request *caam_req = ablkcipher_request_ctx(req);
-	int ivsize = crypto_ablkcipher_ivsize(ablkcipher);
 	int ret;
 
 	/* allocate extended descriptor */
@@ -1960,6 +1959,7 @@ static int ablkcipher_decrypt(struct ablkcipher_request *req)
 	struct crypto_ablkcipher *ablkcipher = crypto_ablkcipher_reqtfm(req);
 	struct caam_ctx *ctx = crypto_ablkcipher_ctx(ablkcipher);
 	struct caam_request *caam_req = ablkcipher_request_ctx(req);
+	int ivsize = crypto_ablkcipher_ivsize(ablkcipher);
 	int ret;
 
 	/* allocate extended descriptor */
-- 
2.17.0

